### PR TITLE
Add specialized error messages for AOT validation failures

### PR DESCRIPTION
**Pull request by dsouzai**

### CODE REVIEW

**Comment by dsouzai** [comment]
@pshipton @mpirvu could you please review

File: runtime/nls/jitm/j9jit.nls [hunk 1]
```diff
@@ -96,6 +96,159 @@ J9NLS_RELOCATABLE_CODE_NOT_AVAILABLE_WITH_FSD_JVMPI.user_response=Disable JVMPI
...
+
+J9NLS_RELOCATABLE_CODE_METHOD_TRAMPOLINE_MISMATCH=Method Trampoline feature mismatch. Ignoring AOT code in shared class cache.
+# START NON-TRANSLATABLE
+J9NLS_RELOCATABLE_CODE_METHOD_TRAMPOLINE_MISMATCH.explanation=The JVM is not running in the same Method Trampoline mode as the one that generated the code in the SCC.
+J9NLS_RELOCATABLE_CODE_METHOD_TRAMPOLINE_MISMATCH.system_action=The compiler will not generate AOT code. AOT code from the shared class cache will not be used.
+J9NLS_RELOCATABLE_CODE_METHOD_TRAMPOLINE_MISMATCH.user_response=Destroy and recreate the AOT code in the shared class cache for the current platform using the -Xshareclasses:reset option.
+J9NLS_RELOCATABLE_CODE_METHOD_TRAMPOLINE_MISMATCH.link=dita:///diag/appendixes/cmdline/cmdline_x.dita#cmdline_x/xshareclasses-reset
+# END NON-TRANSLATABLE
+
+J9NLS_RELOCATABLE_CODE_MULTI_TENANCY_MISMATCH=Multi Tenancy feature mismatch. Ignoring AOT code in shared class cache.
```
Comment (pshipton) [inline runtime/nls/jitm/j9jit.nls:163]: Not sure this is needed since it's not supported any more. I think I saw some of the JIT code being deleted.

**Comment by pshipton** [comment]
lgtm

File: runtime/nls/jitm/j9jit.nls (hunk 1 above)
Comment (mpirvu) [reply runtime/nls/jitm/j9jit.nls:163]: Indeed, scanning the code I cannot find a point where we set this flag, so this particular flag can be deleted.

Comment (dsouzai) [reply runtime/nls/jitm/j9jit.nls:163]: Ok I'll remove this.

**Comment by dsouzai** [comment]
@pshipton @mpirvu changes made.

**Comment by mpirvu** [comment]
jenkins test sanity all jdk11

**Comment by mpirvu** [comment]
All tests have passed, hence merging

**Pull request merged**